upup/pkg/fi/nodeup/nodetasks/archive.go

```diff
@@ -0,0 +1,198 @@
+
+	// TargetDir is the directory for extraction
+	TargetDir string `json:"target,omitempty"`
+}
+
+const (
+	localArchiveDir      = "/var/cache/nodeup/archives/"
+	localArchiveStateDir = "/var/cache/nodeup/archives/state/"
+)
+
+var _ fi.HasDependencies = &Archive{}
+
+// GetDependencies implements HasDependencies::GetDependencies
+func (e *Archive) GetDependencies(tasks map[string]fi.Task) []fi.Task {
+	var deps []fi.Task
+
+	// Requires parent directories to be created
+	for _, v := range findCreatesDirParents(e.TargetDir, tasks) {
+		deps = append(deps, v)
+	}
+
+	return deps
+}
+
+var _ fi.HasName = &Archive{}
+
+func (e *Archive) GetName() *string {
+	return &e.Name
+}
+
+func (e *Archive) SetName(name string) {
+	e.Name = name
+}
+
+// String returns a string representation, implementing the Stringer interface
+func (e *Archive) String() string {
+	return fmt.Sprintf("Archive: %s %s->%s", e.Name, e.Source, e.TargetDir)
+}
+
+var _ CreatesDir = &Archive{}
+
+// Dir implements CreatesDir::Dir
+func (e *Archive) Dir() string {
+	return e.TargetDir
+}
+
+func (e *Archive) Find(c *fi.Context) (*Archive, error) {
+	// We write a marker file to prevent re-execution
+	localStateFile := path.Join(localArchiveStateDir, e.Name)
+	stateBytes, err := ioutil.ReadFile(localStateFile)
+	if err != nil {
+		if os.IsNotExist(err) {
+			stateBytes = nil
+		} else {
+			glog.Warningf("error reading archive state %s: %v", localStateFile, err)
+			// We can just reinstall
+			return nil, nil
+		}
+	}
+
+	if stateBytes == nil {
+		return nil, nil
+	}
+
+	state := &Archive{}
+	if err := json.Unmarshal(stateBytes, state); err != nil {
+		glog.Warningf("error unmarshalling archive state %s: %v", localStateFile, err)
+		// We can just reinstall
+		return nil, nil
+	}
+
+	if state.Hash == e.Hash && state.TargetDir == e.TargetDir {
+		return state, nil
+	}
+	return nil, nil
+}
+
+func (e *Archive) Run(c *fi.Context) error {
+	return fi.DefaultDeltaRunMethod(e, c)
+}
+
+func (_ *Archive) CheckChanges(a, e, changes *Archive) error {
+	return nil
+}
+
+func (_ *Archive) RenderLocal(t *local.LocalTarget, a, e, changes *Archive) error {
+	if a == nil {
+		glog.Infof("Installing archive %q", e.Name)
+
+		localFile := path.Join(localArchiveDir, e.Name)
+		if err := os.MkdirAll(localArchiveDir, 0755); err != nil {
+			return fmt.Errorf("error creating directories %q: %v", localArchiveDir, err)
+		}
+
+		var hash *hashing.Hash
+		if e.Hash != "" {
+			parsed, err := hashing.FromString(e.Hash)
+			if err != nil {
+				return fmt.Errorf("error paring hash: %v", err)
+			}
+			hash = parsed
+		}
+		if _, err := fi.DownloadURL(e.Source, localFile, hash); err != nil {
+			return err
+		}
+
+		targetDir := e.TargetDir
+		if err := os.MkdirAll(targetDir, 0755); err != nil {
+			return fmt.Errorf("error creating directories %q: %v", targetDir, err)
+		}
+
+		args := []string{"tar", "xf", localFile, "-C", targetDir}
+		glog.Infof("running command %s", args)
```

Comment:
shoud we `glog.V(3).Infof("running command %s", args)`?

Reply:
It looks like in nodeup we glog.Infof when we're making changes and glog.V(2).Infof when we're just looking for stuff.  So I think glog.Infof here